### ISSUE TITLE
ksc array from 7 months ago improperly escaped special characters

### ISSUE THREAD

**Issue by xaminmo** [myspaghetti/macos-guest-virtualbox#147]
From this commit and/or its predecessors: https://github.com/myspaghetti/macos-guest-virtualbox/commit/dfc0fd22ce3d219b9194b2b344810662e3068690

**ERRORS**;
```
 ./macos-guest-virtualbox.sh
./macos-guest-virtualbox.sh: line 619: []]=1B 9B: bad array subscript
./macos-guest-virtualbox.sh: line 619: [\]=2B AB: bad array subscript
./macos-guest-virtualbox.sh: line 619: [@]=2A 03 83 AA: invalid associative array key
./macos-guest-virtualbox.sh: line 619: [*]=2A 09 89 AA: invalid associative array key

                Semi-automatic installer of macOS on VirtualBox
-------------------------------------------------------------------------------
```

These lines are incorrectly escaped, and cause script errors at least in cygwin bash:
**CURRENT**:
```
    ["["]="1A 9A"
    ["]"]="1B 9B"
    ['\']="2B AB"
    ["@"]="2A 03 83 AA"
    ["*"]="2A 09 89 AA"
```
Each of those lines needs a backslash.

**CORRECT**:
```
    ["\["]="1A 9A"
    ["\]"]="1B 9B"
    ['\\']="2B AB"
    ["\@"]="2A 03 83 AA"
    ["\*"]="2A 09 89 AA"
```

**Comment by myspaghetti** [comment]
What version of Bash are you using?

**Comment by myspaghetti** [comment]
None of these characters should be escaped on Bash 4. Please run the following script in Bash 4 and report any errors:
[ksc.txt](https://github.com/myspaghetti/macos-guest-virtualbox/files/4052533/ksc.txt)

This is a [possible duplicate of a CentOS bug (issue 126)](https://github.com/myspaghetti/macos-guest-virtualbox/issues/126)

**Comment by myspaghetti** [comment]
None of these characters should be escaped on Bash 4. Please run the following script in Bash 4 and report any errors:

```
declare -A dict=( ['\']="slash" ) # bad array subscript
declare -A dict=( ["]"]="closingbracket" ) # bad array subscript
declare -A dict=( ["@"]="at" ) # invalid associative array key
declare -A dict=( ["*"]="asterisk" ) # invalid associative array key
```

This is a [possible duplicate of a CentOS bug (issue 126)](https://github.com/myspaghetti/macos-guest-virtualbox/issues/126)

**Comment by xaminmo** [comment]
It definitely seemed odd to need to escape anything that was inside of single quotes (other than a single quote).  I did some version testing.

My **failing** laptop with MobaX using Cygwin 2, says 
`GNU bash, version 4.1.17(0)-release (i686-pc-cygwin). `

Testing from a new CygWin 3 install elsewhere, it **works** with 
`GNU bash, version 4.4.12(3)-release (x86_64-unknown-cygwin).`

Running the script on Debian also **works** with 
```GNU bash, version 5.0.3(1)-release (x86_64-pc-linux-gnu)  (no vbox there)
GNU bash, version 4.3.30(1)-release (x86_64-pc-linux-gnu).
GNU bash, version 4.3.30(1)-release (i586-pc-linux-gnu)```

It **fails** on Debian with:
`GNU bash, version 4.2.45(1)-release (x86_64-pc-linux-gnu)`

The most simple fix here is to update the readme and the script dependencies line to say it requires Bash 4.3 or later.  The alternative would be to do the escapes listed in the OP.  They seem to port fine into current Bashes, but I don't know if that really makes sense to adjust things to fix issues in shells unpatched in 5 years.

**Comment by myspaghetti** [comment]
"Fixed" in [version 0.78.1](https://github.com/myspaghetti/macos-guest-virtualbox/commit/68c9fb7f76b89d699b89fac679ed924a64c212b3) by raising Bash version requirement to 4.3

**Comment by xaminmo** [comment]
Thanks!  A++.
